js/validateForm.js

```diff
@@ -0,0 +1,114 @@
+const nameEl = document.querySelector('#name');
+const textEl = document.querySelector('#message');
+const addSmall = document.createElement('small');
+
+const form = document.querySelector('#formContact');
+
+const isRequired = (value) => (value !== '');
+const isBetween = (length, min, max) => (!(length < min || length > max));
+
+const showError = (input, message) => {
+  // get the form-field element
+  const formField = input.parentElement;
+  // add the error class
+  input.classList.remove('success');
+  input.classList.add('error');
+
+  // show the error message
+  formField.appendChild(addSmall);
+  const error = formField.querySelector('small');
+  error.textContent = message;
+};
+
+const showSuccess = (input) => {
+  // get the form-field element
+  const formField = input.parentElement;
+
+  // remove the error class
+  input.classList.add('success');
+  input.classList.remove('error');
+
+  if (formField.querySelector('small' !== '')) {
+    const deleteSmall = formField.querySelector('small');
+    deleteSmall.remove();
+  }
+};
+
+const checkName = () => {
+  let valid = false;
+
+  const min = 3; const max = 25;
+
+  const name = nameEl.value.trim();
+
+  if (!isRequired(name)) {
+    showError(nameEl, 'Username cannot be blank.');
+  } else if (!isBetween(name.length, min, max)) {
+    showError(nameEl, `Name must be between ${min} and ${max} characters.`);
+  } else {
+    showSuccess(nameEl);
+    valid = true;
+  }
+  return valid;
+};
+
+const checkText = () => {
+  let valid = false;
+
+  const min = 3; const max = 250;
+
+  const text = textEl.value.trim();
+
+  if (!isRequired(text)) {
+    showError(textEl, 'Username cannot be blank.');
+  } else if (!isBetween(text.length, min, max)) {
+    showError(textEl, `Text must be between ${min} and ${max} characters.`);
+  } else {
+    showSuccess(textEl);
+    valid = true;
+  }
+  return valid;
+};
+
+form.addEventListener('click', (e) => {
+  // validate fields
+  const isUsernameValid = checkName();
+  const isTextValid = checkText();
+
+  const isFormValid = isUsernameValid
+        && isTextValid;
+
+  // submit to the server if the form is valid
+  if (isFormValid === false) {
+    // prevent the form from submitting
+    e.preventDefault();
+  }
+});
+
+const debounce = (fn, delay = 500) => {
+  let timeoutId;
+  return (...args) => {
+    // cancel the previous timer
+    if (timeoutId) {
+      clearTimeout(timeoutId);
+    }
+    // setup a new timer
+    timeoutId = setTimeout(() => {
+      // eslint-disable-next-line prefer-spread
+      fn.apply(null, args);
+    }, delay);
+  };
+};
+
+form.addEventListener('input', debounce((e) => {
+  switch (e.target.id) {
+    case 'name':
+      checkName();
+      break;
+    case 'message':
+      checkText();
+      break;
+    default:
+      break;
+  }
+}));
```

Comment:
- [ ] Currently, you do not have any validation for email. The user can submit an email in uppercase which should not be allowed according to the requirement. Also the user can submit an email without the `@` sign which should not be allowed as well

Reply:
- [ ] You have an error that comes from Card.js file kindly check that 
![image](https://user-images.githubusercontent.com/62705947/218130530-ddfc7dce-482f-4a17-bca8-1c523c2ee2b4.png)

Reply:
- [ ] Hi @mdg0410, kindly check project requirements, the form should not be able to submit when error, and yours do submit. The error message should show around the button get in touch, which yours doesn't 
![image](https://user-images.githubusercontent.com/62705947/218134630-073e8afb-cf88-4eaf-8142-975c60b4e3a4.png)

Reply:
# Thanks @GzimAsani 😎🤘,
If it wasn't for your comment I wouldn't have been motivated to improve my code and put more conscience in checking better the future requirements.

Good luck and thanks again. 🍀

Reply:
> # Thanks @GzimAsani 😎🤘,
> If it wasn't for your comment I wouldn't have been motivated to improve my code and put more conscience in checking better the future requirements.
> 
> Good luck and thanks again. 🍀

I am glad that I could help and I am sorry if I wasn’t clear enough the first time 🙏

Reply:
Do not worry 👽